build/src/main.rs

```diff
@@ -79,17 +86,48 @@ fn main() {
 			.help("Injects RUNTIME_VERSION global export")
 			.takes_value(true)
 			.long("runtime-version"))
+		.arg(Arg::with_name("source_target")
+			.help("Cargo target type kind (wasm32-unknown-unknown/emscripten)")
+			.takes_value(true)
+			.long("target"))
+		.arg(Arg::with_name("final_name")
+			.help("Final wasm binary name")
+			.takes_value(true)
+			.long("final"))
+		.arg(Arg::with_name("save_raw")
+			.help("Save intermediate raw bytecode to path")
+			.takes_value(true)
+			.long("save-raw"))
 		.get_matches();
 
     let target_dir = matches.value_of("target").expect("is required; qed");
     let wasm_binary = matches.value_of("wasm").expect("is required; qed");
+	let mut source_input = source::SourceInput::new(target_dir, wasm_binary);
 
-	process_output(target_dir, wasm_binary).expect("Failed to process cargo target directory");
+	let source_target_val = matches.value_of("source_target").unwrap_or_else(|| source::EMSCRIPTEN_PATH);
+	if source_target_val == source::UNKNOWN_PATH {
+		source_input = source_input.unknown()
+	} else if source_target_val == source::EMSCRIPTEN_PATH {
```

Comment:
why it is called `PATH`?

Reply:
because it's path were build output is located

Reply:
Then it seems a little confusing why is comparison between compilation target triplet and path component takes place.

Reply:
until you know they match

Reply:
anyway, renamed it